### ISSUE TITLE
Lettuce 6.0.1 fails with GraalVM 20.3

### ISSUE THREAD

**Issue by atrianac** [lettuce-io/lettuce-core#1562]
## Bug Report

A `java.lang.NoSuchMethodError: io.lettuce.core.sentinel.api.sync.RedisSentinelCommands.dispatch` is raised with GraalVM native image 20.3

#### Current Behavior

<details>
<summary>Stack trace</summary>

```java
java.lang.InternalError: java.lang.NoSuchMethodError: io.lettuce.core.sentinel.api.sync.RedisSentinelCommands.dispatch(io.lettuce.core.protocol.ProtocolKeyword, io.lettuce.core.output.CommandOutput, io.lettuce.core.protocol.CommandArgs)
        at java.lang.reflect.Proxy.newProxyInstance(Proxy.java:1030)
        at java.lang.reflect.Proxy.newProxyInstance(Proxy.java:1008)
        at io.lettuce.core.RedisChannelHandler.syncHandler(RedisChannelHandler.java:314)
        at io.lettuce.core.sentinel.StatefulRedisSentinelConnectionImpl.<init>(StatefulRedisSentinelConnectionImpl.java:55)
        at io.lettuce.core.RedisClient.newStatefulRedisSentinelConnection(RedisClient.java:643)
        at io.lettuce.core.RedisClient.doConnectSentinelAsync(RedisClient.java:568)
        at io.lettuce.core.RedisClient.lambda$connectSentinelAsync$5(RedisClient.java:513)
        at reactor.core.publisher.Mono.lambda$fromCompletionStage$0(Mono.java:430)
        at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:44)
        at reactor.core.publisher.Mono.subscribe(Mono.java:4213)
        at reactor.core.publisher.Mono.subscribeWith(Mono.java:4324)
        at reactor.core.publisher.Mono.toFuture(Mono.java:4658)
        at io.lettuce.core.RedisClient.connectSentinelAsync(RedisClient.java:544)
        at io.lettuce.core.RedisClient.connectSentinelAsync(RedisClient.java:489)
        at io.lettuce.core.masterreplica.SentinelTopologyProvider.getNodesAsync(SentinelTopologyProvider.java:95)
        at io.lettuce.core.masterreplica.UpstreamReplicaTopologyRefresh.getNodes(UpstreamReplicaTopologyRefresh.java:73)
        at io.lettuce.core.masterreplica.SentinelConnector.connectAsync(SentinelConnector.java:68)
        at io.lettuce.core.masterreplica.MasterReplica.connectAsyncSentinelOrStaticSetup(MasterReplica.java:224)
        at io.lettuce.core.masterreplica.MasterReplica.connectAsync(MasterReplica.java:202)
        at dev.profunktor.redis4cats.connection.RedisMasterReplica$.$anonfun$acquireAndRelease$1(RedisMasterReplica.scala:50)
        at zio.internal.FiberContext.evaluateNow(FiberContext.scala:361)
        at zio.internal.FiberContext.$anonfun$evaluateLater$1(FiberContext.scala:778)
        at cats.effect.internals.PoolUtils$$anon$2$$anon$3.run(PoolUtils.scala:52)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.lang.Thread.run(Thread.java:834)
        at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:519)
        at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
Caused by: java.lang.NoSuchMethodError: io.lettuce.core.sentinel.api.sync.RedisSentinelCommands.dispatch(io.lettuce.core.protocol.ProtocolKeyword, io.lettuce.core.output.CommandOutput, io.lettuce.core.protocol.CommandArgs)
        at com.sun.proxy.$Proxy141.<clinit>(Unknown Source)
        at com.oracle.svm.core.classinitialization.ClassInitializationInfo.invokeClassInitializer(ClassInitializationInfo.java:351)
        at com.oracle.svm.core.classinitialization.ClassInitializationInfo.initialize(ClassInitializationInfo.java:271)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at java.lang.reflect.Proxy.newProxyInstance(Proxy.java:1022)
        ... 27 more
```

</details>

#### Input Code

The reflect-config.json looks lke this: 

<details>
<summary>Input Code</summary>

```json
[
  {
    "name": "org.slf4j.impl.StaticLoggerBinder",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.DateConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.MessageConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.ThrowableProxyConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.NopThrowableInformationConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.ContextNameConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.core.pattern.color.BoldYellowCompositeConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.LoggerConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.core.pattern.ReplacingCompositeConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.core.pattern.color.BoldBlueCompositeConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.core.pattern.color.CyanCompositeConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.core.pattern.color.RedCompositeConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.core.pattern.color.WhiteCompositeConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.PropertyConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.ExtendedThrowableProxyConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.RootCauseFirstThrowableProxyConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.MethodOfCallerConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.LevelConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.core.pattern.IdentityCompositeConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.core.pattern.color.BoldWhiteCompositeConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.MarkerConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.core.pattern.color.BoldCyanCompositeConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.core.pattern.color.BoldMagentaCompositeConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.RelativeTimeConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.core.pattern.color.MagentaCompositeConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.ClassOfCallerConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.LineOfCallerConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.FileOfCallerConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.core.pattern.color.BoldGreenCompositeConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.LocalSequenceNumberConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.core.pattern.color.YellowCompositeConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.ExtendedThrowableProxyConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.color.HighlightingCompositeConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.core.pattern.color.GrayCompositeConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.MDCConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.ClassOfCallerConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.core.pattern.color.BoldRedCompositeConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.core.pattern.color.GreenCompositeConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.core.pattern.color.BlackCompositeConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.ThreadConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.pattern.LineSeparatorConverter",
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.classic.encoder.PatternLayoutEncoder",
    "allPublicMethods": true,
    "allDeclaredConstructors": true
  },
  {
    "name": "ch.qos.logback.core.ConsoleAppender",
    "allPublicMethods": true,
    "allDeclaredConstructors": true
  },
  {
    "name": "com.sun.org.apache.xerces.internal.jaxp.SAXParserFactoryImpl",
    "allDeclaredConstructors": true
  },
  {
    "name":"io.netty.util.internal.shaded.org.jctools.queues.BaseMpscLinkedArrayQueueColdProducerFields",
    "fields":[{"name":"producerLimit","allowUnsafeAccess" :  true}]
  },
  {
    "name":"io.netty.util.internal.shaded.org.jctools.queues.BaseMpscLinkedArrayQueueConsumerFields",
    "fields":[{"name":"consumerIndex","allowUnsafeAccess" :  true}]
  },
  {
    "name":"io.netty.util.internal.shaded.org.jctools.queues.BaseMpscLinkedArrayQueueProducerFields",
    "fields":[{"name":"producerIndex", "allowUnsafeAccess" :  true}]
  },
  {
    "name":"io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueConsumerIndexField",
    "fields":[{"name":"consumerIndex", "allowUnsafeAccess" :  true}]
  },
  {
    "name":"io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueProducerIndexField",
    "fields":[{"name":"producerIndex", "allowUnsafeAccess" :  true}]
  },
  {
    "name":"io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueProducerLimitField",
    "fields":[{"name":"producerLimit","allowUnsafeAccess" :  true}]
  },
  {
    "name":"java.nio.Buffer",
    "fields":[{"name":"address", "allowUnsafeAccess":true}]
  },
  {
    "name":"java.nio.DirectByteBuffer",
    "fields":[{"name":"cleaner", "allowUnsafeAccess":true}],
    "methods":[{"name":"<init>","parameterTypes":["long","int"] }]
  },
  {
    "name":"io.netty.buffer.AbstractReferenceCountedByteBuf",
    "fields":[{"name":"refCnt", "allowUnsafeAccess":true}]
  },
  {
    "name":"io.netty.buffer.AbstractByteBufAllocator",
    "allPublicMethods": true,
    "allDeclaredFields":true,
    "allDeclaredMethods":true,
    "allDeclaredConstructors":true
  },
  {
    "name":"io.netty.buffer.PooledByteBufAllocator",
    "allPublicMethods": true,
    "allDeclaredFields":true,
    "allDeclaredMethods":true,
    "allDeclaredConstructors":true
  },
  {
    "name":"io.netty.channel.ChannelDuplexHandler",
    "allPublicMethods": true,
    "allDeclaredConstructors":true
  },
  {
    "name":"io.netty.channel.ChannelHandlerAdapter",
    "allPublicMethods": true,
    "allDeclaredConstructors":true
  },
  {
    "name": "io.netty.channel.ChannelInboundHandlerAdapter",
    "allPublicMethods": true,
    "allDeclaredConstructors":true
  },
  {
    "name": "io.netty.channel.ChannelInitializer",
    "allPublicMethods": true,
    "allDeclaredConstructors":true
  },
  {
    "name": "io.netty.channel.ChannelOutboundHandlerAdapter",
    "allPublicMethods": true,
    "allDeclaredConstructors":true
  },
  {
    "name": "io.netty.channel.DefaultChannelPipeline$HeadContext",
    "allPublicMethods": true,
    "allDeclaredConstructors":true
  },
  {
    "name": "io.netty.channel.DefaultChannelPipeline$TailContext",
    "allPublicMethods": true,
    "allDeclaredConstructors":true
  },
  {
    "name": "io.netty.channel.socket.nio.NioSocketChannel",
    "allPublicMethods": true,
    "allDeclaredConstructors":true
  },
  {
    "name": "io.netty.handler.codec.MessageToByteEncoder",
    "allPublicMethods": true,
    "allDeclaredConstructors":true
  },
  {
    "name":"io.netty.util.ReferenceCountUtil",
    "allPublicMethods": true,
    "allDeclaredConstructors":true
  },
  {
    "name": "org.HdrHistogram.Histogram"
  },
  {
    "name": "org.LatencyUtils.PauseDetector"
  }
]
```

</details>

#### Expected behavior/code

The application runs appropriately. 

#### Environment

- Lettuce version(s): 6.0.1
- Redis version: 6.0

#### Additional context

We are using redis4cats as Scala wrapper

**Comment by mp911de** [comment]
We introduced `dispatch(…)` methods on `RedisSentinelCommands` with Lettuce 6.0.2. Care to upgrade to see whether the issue goes away with the newer version? Beyond that, I don't have the bandwidth to investigate GraalVM issues. Feel free to submit a pull request it you can fix the issue.

**Comment by atrianac** [comment]
> We introduced `dispatch(…)` methods on `RedisSentinelCommands` with Lettuce 6.0.2. Care to upgrade to see whether the issue goes away with the newer version? Beyond that, I don't have the bandwidth to investigate GraalVM issues. Feel free to submit a pull request it you can fix the issue.

Thanks for the reply and the context. I will do :)

**Comment by mp911de** [comment]
Thanks a lot, that fixed with #1563.
